packages/admin-next/dashboard/src/v2-routes/shipping/shipping-options-create/components/create-shipping-options-form/create-shipping-options-form.tsx

```diff
@@ -0,0 +1,378 @@
+import { useForm } from "react-hook-form"
+import { zodResolver } from "@hookform/resolvers/zod"
+import React, { useEffect } from "react"
+import * as zod from "zod"
+
+import {
+  Button,
+  Heading,
+  Input,
+  ProgressStatus,
+  ProgressTabs,
+  RadioGroup,
+  Switch,
+  Text,
+  clx,
+  Select,
+} from "@medusajs/ui"
+import { ServiceZoneDTO } from "@medusajs/types"
+import { useTranslation } from "react-i18next"
+
+import {
+  RouteFocusModal,
+  useRouteModal,
+} from "../../../../../components/route-modal"
+import { Form } from "../../../../../components/common/form"
+import { CreateShippingOptionsPricesForm } from "./create-shipping-options-prices-form"
+import { useCreateShippingOptions } from "../../../../../hooks/api/shipping-options"
+import { useShippingProfiles } from "../../../../../hooks/api/shipping-profiles"
+import { getDbAmount } from "../../../../../lib/money-amount-helpers"
+
+enum Tab {
+  DETAILS = "details",
+  PRICING = "pricing",
+}
+
+enum ShippingAllocation {
+  FlatRate = "flat",
+  Calculated = "calculated",
+}
+
+type StepStatus = {
+  [key in Tab]: ProgressStatus
+}
+
+const CreateServiceZoneSchema = zod.object({
+  name: zod.string().min(1),
+  price_type: zod.nativeEnum(ShippingAllocation),
+  enable_in_store: zod.boolean().optional(),
+  shipping_profile_id: zod.string(),
+  region_prices: zod.record(zod.string(), zod.string().optional()),
+  currency_prices: zod.record(zod.string(), zod.string().optional()),
+})
+
+type CreateServiceZoneFormProps = {
+  zone: ServiceZoneDTO
+}
+
+export function CreateShippingOptionsForm({
+  zone,
+}: CreateServiceZoneFormProps) {
+  const { t } = useTranslation()
+  const { handleSuccess } = useRouteModal()
+  const [tab, setTab] = React.useState<Tab>(Tab.DETAILS)
```

Comment:
should be possible to go back and forth between tabs once the details page has been validated.